front/lib/resources/webhook_request_resource.ts

```diff
@@ -133,6 +146,93 @@ export class WebhookRequestResource extends BaseResource<WebhookRequestModel> {
     }
   }
 
+  static async getWorkspaceIdsWithTooManyRequests({
+    webhookRequestTtl = WEBHOOK_REQUEST_TTL,
+    maxWebhookRequestsToKeep = MAX_WEBHOOK_REQUESTS_TO_KEEP,
+  }: Partial<CleanUpWorkspaceOptions> = {}) {
+    // eslint-disable-next-line dust/no-raw-sql
+    const rows = await frontSequelize.query<{
+      workspaceId: ModelId;
+      total_entries: number;
+      oldest_created_at: Date;
+    }>(
+      `
+      SELECT
+        "workspaceId",
+          COUNT(*) AS "total_entries",
+          MIN("createdAt") AS "oldest_created_at"
+      FROM public.webhook_requests
+      GROUP BY "workspaceId"
+      HAVING
+        COUNT(*) > :max_webhook_requests_to_keep
+        OR MIN("createdAt") < now() - interval :webhook_request_ttl
+      ORDER BY "workspaceId" ASC;
+    `,
+      {
+        type: QueryTypes.SELECT,
+        replacements: {
+          max_webhook_requests_to_keep: maxWebhookRequestsToKeep,
+          webhook_request_ttl: webhookRequestTtl,
+        },
+      }
+    );
+
+    return rows.map((row) => row.workspaceId);
+  }
+
+  static async cleanUpWorkspace(
+    auth: Authenticator,
+    {
+      webhookRequestTtl = WEBHOOK_REQUEST_TTL,
+      maxWebhookRequestsToKeep = MAX_WEBHOOK_REQUESTS_TO_KEEP,
+    }: Partial<CleanUpWorkspaceOptions> = {}
+  ) {
+    const oldRequests = await this.baseFetch(auth, {
+      where: {
+        createdAt: {
+          [Op.lt]: literal(`now() - interval '${webhookRequestTtl}'`),
+        },
+      },
+    });
+
+    logger.info(
+      {
+        toDelete: oldRequests.length,
+        workspaceId: auth.getNonNullableWorkspace().sId,
+      },
+      "Cleaning up old webhook requests"
+    );
+
+    await concurrentExecutor(
+      oldRequests,
+      async (request) => {
+        await request.delete(auth);
+      },
+      { concurrency: 16 }
+    );
```

Comment:
can't we do a bulkDelete on request resource and pass a list of ids ? this does a lot less db calls no ?